From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 14 Dec 2025 23:19:27 +0800
Subject: [PATCH] HC: Support reuse SDL-window for LoaderMod

---
 .../src/arc/backend/sdl/SdlApplication.java       | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/backends/backend-sdl/src/arc/backend/sdl/SdlApplication.java b/backends/backend-sdl/src/arc/backend/sdl/SdlApplication.java
index f356fe21605f785cc84845c75a2c26568a9b5782..8248211d452110c3e158fe6cd3b1f995c775b38e 100644
--- a/backends/backend-sdl/src/arc/backend/sdl/SdlApplication.java
+++ b/backends/backend-sdl/src/arc/backend/sdl/SdlApplication.java
@@ -112,6 +112,21 @@ public class SdlApplication implements Application{
 
         if(OS.isMac) restartMac();
 
+        //MDTX: Special handle for Loader
+        if(System.getProperty("MDTX-SDL-window") != null){
+            Log.info("[Core] Initialize reusing window and context.");
+            window = Long.parseLong(System.getProperty("MDTX-SDL-window"));
+            context = Long.parseLong(System.getProperty("MDTX-SDL-context"));
+
+            config.width = Integer.parseInt(System.getProperty("MDTX-SDL-width"));
+            config.height = Integer.parseInt(System.getProperty("MDTX-SDL-height"));
+
+            int[] ver = new int[3];
+            SDL_GetVersion(ver);
+            Log.info("[Core] Initialized SDL v@.@.@", ver[0], ver[1], ver[2]);
+            return;
+        }
+
         check(SDL_Init(SDL_INIT_VIDEO | SDL_INIT_EVENTS));
 
         //show native IME candidate UI
