From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Fri, 19 Apr 2024 12:27:50 +0800
Subject: [PATCH] OC: no MapPreviewLoader.setupLoaders

---
 core/src/mindustry/ClientLauncher.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/src/mindustry/ClientLauncher.java b/core/src/mindustry/ClientLauncher.java
index 227495e812499ffaae06c12f6fe235b68feab488..a91ea8917a6c687d728e5d6fe1e39f2db53f2c87 100644
--- a/core/src/mindustry/ClientLauncher.java
+++ b/core/src/mindustry/ClientLauncher.java
@@ -140,7 +140,8 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
         assets.load("sprites/error.png", Texture.class);
         atlas = TextureAtlas.blankAtlas();
         Vars.net = new Net(platform.getNet());
-        MapPreviewLoader.setupLoaders();
+        //MDTX: optimize performance (inspired by Foo)
+//        MapPreviewLoader.setupLoaders();
         mods = new Mods();
         schematics = new Schematics();
 
