From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 15 Sep 2024 15:53:45 +0800
Subject: [PATCH] FC: reliableSync

---
 core/src/mindustry/net/Net.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/src/mindustry/net/Net.java b/core/src/mindustry/net/Net.java
index 2c297279bf049fe625f8b1d39d7181a6c2d54d74..3a9d6f5d55bd526c4ca5304a4e59e1d53e0c552a 100644
--- a/core/src/mindustry/net/Net.java
+++ b/core/src/mindustry/net/Net.java
@@ -243,6 +243,7 @@ public class Net{
                 con.send(object, reliable);
             }
         }else{
+            reliable |= LogicExt.reliableSync.get();
             provider.sendClient(object, reliable);
         }
     }
