From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 15 Sep 2024 15:53:45 +0800
Subject: [PATCH] FC: reliableSync

---
 core/src/mindustry/net/Net.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/src/mindustry/net/Net.java b/core/src/mindustry/net/Net.java
index 22b66b83cde27a73415b9154ed813cdfeb3d3b36..a0ffd0d960772e45f16d935c46ffdcfcfaded35e 100644
--- a/core/src/mindustry/net/Net.java
+++ b/core/src/mindustry/net/Net.java
@@ -231,6 +231,7 @@ public class Net{
                 con.send(object, reliable);
             }
         }else{
+            reliable |= LogicExt.reliableSync;
             provider.sendClient(object, reliable);
         }
     }
